### ISSUE TITLE
Underscores for SRV queries

### ISSUE THREAD

**Issue by wilem82** [ogham/dog#54]
Hi, is there a limitation in `dog` that doesn't allow this:

    dog -t SRV _servicename._tcp.example.com

Or am I using it wrong? `dog` gives the error:

    Invalid options: Invalid domain "_servicename"

But underscores are a normal part of SRV queries, from what I gathered by reading https://tools.ietf.org/html/rfc2782. Thank you.

**Comment by cocoonkid** [comment]
yes, i wonder about the same. are SRV queries TLSA and CNAME possible?

**Comment by gbe0** [comment]
Same problem for me, trying to lookup some TXT record that are included for a SPF record which contain underscores.

**Comment by nullvariable** [comment]
yeah, `dog CNAME _acme-challenge.xxxxx` reports the domain as invalid, but the `_` is valid for CNAMEs (but only when the CNAME is not being treated as a host name, similar to a TXT record)

I found this answer helpful in understanding the nuance here:
https://serverfault.com/a/928204/196152

The TLDR summary is that hostnames are greatly restricted in valid characters, but domain names are not. A CNAME record can be both, so dog should not validate these the same way as A records.
